src/map/script.cpp

```diff
@@ -20381,6 +20381,58 @@ BUILDIN_FUNC(instance_info)
 	return SCRIPT_CMD_SUCCESS;
 }
 
+/*------------------------------------------
+*instance_live_info( <info type>{, <instance id>} );
+- ILI_NAME : Instance Name
+- ILI_MODE : Instance Mode options (IM_NONE, IM_CHAR etc..)
+- ILI_OWNER : owner id (if any)
+*------------------------------------------*/
+BUILDIN_FUNC(instance_live_info)
+{
+	struct instance_db *db = NULL;
+	struct instance_data *im = NULL;
+
+	int type = script_getnum(st, 2);
+	int id = 0;
+
+	if (script_hasdata(st, 3) == NULL)
+		id = script_instancegetid(st);
+	else
+		id = script_getnum(st, 3);
+
+	if (id) {
+		im = &instance_data[id];
```

Comment:
it's a bit unsafe here. if `id` is defined in script, but no range check at least for `if (id < 1 || id >= MAX_INSTANCE_DATA)`

Reply:
Good catch!